src/mame/heathkit/tlb.cpp

```diff
@@ -0,0 +1,749 @@
+	PORT_BIT(0x040, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_NAME("7")          PORT_CODE(KEYCODE_7)          PORT_CHAR('7') PORT_CHAR('&')
+	PORT_BIT(0x080, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_NAME("8")          PORT_CODE(KEYCODE_8)          PORT_CHAR('8') PORT_CHAR('*')
+	PORT_BIT(0x100, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_NAME("9")          PORT_CODE(KEYCODE_9)          PORT_CHAR('9') PORT_CHAR('(')
+	PORT_BIT(0x200, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_NAME("Esc")        PORT_CODE(KEYCODE_ESC)        PORT_CHAR(27)
+
+	PORT_START("SW401")
+	PORT_DIPNAME( 0x0f, 0x0c, "Baud Rate")          PORT_DIPLOCATION("SW401:1,2,3,4")
+	PORT_DIPSETTING(    0x01, "110")
+	PORT_DIPSETTING(    0x02, "150")
+	PORT_DIPSETTING(    0x03, "300")
+	PORT_DIPSETTING(    0x04, "600")
+	PORT_DIPSETTING(    0x05, "1200")
+	PORT_DIPSETTING(    0x06, "1800")
+	PORT_DIPSETTING(    0x07, "2000")
+	PORT_DIPSETTING(    0x08, "2400")
+	PORT_DIPSETTING(    0x09, "3600")
+	PORT_DIPSETTING(    0x0a, "4800")
+	PORT_DIPSETTING(    0x0b, "7200")
+	PORT_DIPSETTING(    0x0c, "9600")
+	PORT_DIPSETTING(    0x0d, "19200")
+	PORT_DIPNAME( 0x30, 0x00, "Parity")              PORT_DIPLOCATION("SW401:5,6")
+	PORT_DIPSETTING(    0x00, DEF_STR(None))
+	PORT_DIPSETTING(    0x10, "Odd")
+	PORT_DIPSETTING(    0x20, "None")
+	PORT_DIPSETTING(    0x30, "Even")
+	PORT_DIPNAME( 0x40, 0x00, "Parity Type")         PORT_DIPLOCATION("SW401:7")
+	PORT_DIPSETTING(    0x00, DEF_STR(Normal))
+	PORT_DIPSETTING(    0x40, "Stick")
+	PORT_DIPNAME( 0x80, 0x80, "Duplex")              PORT_DIPLOCATION("SW401:8")
+	PORT_DIPSETTING(    0x00, "Half")
+	PORT_DIPSETTING(    0x80, "Full")
+
+	PORT_START("SW402") // stored at 40C8
+	PORT_DIPNAME( 0x01, 0x00, "Cursor")             PORT_DIPLOCATION("SW402:1")
+	PORT_DIPSETTING(    0x00, "Underline")
+	PORT_DIPSETTING(    0x01, "Block")
+	PORT_DIPNAME( 0x02, 0x00, "Keyclick")           PORT_DIPLOCATION("SW402:2")
+	PORT_DIPSETTING(    0x02, DEF_STR(No))
+	PORT_DIPSETTING(    0x00, DEF_STR(Yes))
+	PORT_DIPNAME( 0x04, 0x00, "Wrap at EOL")        PORT_DIPLOCATION("SW402:3")
+	PORT_DIPSETTING(    0x00, DEF_STR(No))
+	PORT_DIPSETTING(    0x04, DEF_STR(Yes))
+	PORT_DIPNAME( 0x08, 0x00, "Auto LF on CR")      PORT_DIPLOCATION("SW402:4")
+	PORT_DIPSETTING(    0x00, DEF_STR(No))
+	PORT_DIPSETTING(    0x08, DEF_STR(Yes))
+	PORT_DIPNAME( 0x10, 0x00, "Auto CR on LF")      PORT_DIPLOCATION("SW402:5")
+	PORT_DIPSETTING(    0x00, DEF_STR(No))
+	PORT_DIPSETTING(    0x10, DEF_STR(Yes))
+	PORT_DIPNAME( 0x20, 0x00, "Mode")               PORT_DIPLOCATION("SW402:6")
+	PORT_DIPSETTING(    0x00, "Heath/VT52")
+	PORT_DIPSETTING(    0x20, "ANSI")
+	PORT_DIPNAME( 0x40, 0x00, "Keypad Shifted")     PORT_DIPLOCATION("SW402:7")
+	PORT_DIPSETTING(    0x00, DEF_STR(No))
+	PORT_DIPSETTING(    0x40, DEF_STR(Yes))
+	PORT_DIPNAME( 0x80, 0x00, "Refresh")            PORT_DIPLOCATION("SW402:8")
+	PORT_DIPSETTING(    0x00, "50Hz")
+	PORT_DIPSETTING(    0x80, "60Hz")
+INPUT_PORTS_END
+
+
+static INPUT_PORTS_START( super19 )
```

Comment:
If I used the alternative BIOS option, is there a way to modify the port settings like this? I couldn't find anything in the existing code base that did it. I'm currently working on the H89 code and running into the same issue that different ROMs have different dipswitch settings. I would like to try using alternate ROMs there, but don't see a solution to this. If there is not currently a solution, we may want to look at how we can update the framework to allow it.

Reply:
So the parent device would define all the different variants for the dipswitch settings for all child devices? Is there some existing code you can point me to, that has different dipswitch settings for different ROMs? If possible, I'd like to hold off on making changes related to alt bios on this PR.

Reply:
It’ll be OK as-is for now.